src/penalty_parameter_functions.jl

```diff
@@ -83,6 +83,13 @@ function process_penalty_subproblem(r::AbstractPenaltyParameter,
     throw(UnimplementedError("process_penalty_subproblem is unimplemented for penalty parameter of type $(typeof(r))."))
 end
 
+"""
+Resets any mapping of consensus variable ids to penalty parameter values which `r` may have. This avoids potential bugs caused by the same instance of a penalty parameter being used for several calls to `solve`.
+"""
+function reset_penalty_parameter(r::AbstractPenaltyParameter)::Nothing
```

Comment:
Instead of this throwing an error, you can make it the default behavior for the interface. That is have it be
```
function reset_penalty_parameter(r::AbstractPenaltyParameter)::Nothing
    return # intentional no-op
end
```
Then this only needs to be implemented if required.